code/backend/batch/utilities/helpers/embedders/push_embedder.py

```diff
@@ -85,6 +88,32 @@ def __embed(
             logger.error("Failed to upload documents to search index")
             raise Exception(response)
 
+    def generate_image_caption(self, source_url):
```

Comment:
Can we make this "private"? https://www.geeksforgeeks.org/private-methods-in-python/

Just a nit so can tackle it later.